request.go

```diff
@@ -14,16 +15,18 @@ type Request struct {
 	Body     interface{}
 	Puzzled  bool
 	LogRaw   bool
+	Context  context.Context
 }
 
 // NewRequest returns a pointer to a new created Request
-func NewRequest(method, endpoint string, body interface{}) *Request {
+func NewRequest(ctx context.Context, method, endpoint string, body interface{}) *Request {
 	r := Request{
 		Method:   method,
 		Endpoint: endpoint,
 		Header:   make(http.Header),
 		Values:   make(url.Values),
 		Body:     body,
+		Context:  ctx,
```

Comment:
There is no need to have it in our Request object, but it is mirroring the http Request object by putting it in, so it seems like a sensible place

Reply:
ah there is a gotcha here, the context on an http.Request is only populated on a received request not an out going request.  I believe it's there for when connections drop/timeout so that you can stop processing early once you know you won't be writing a response.